### PR TITLE
Optimized the reading performance of Duplicate tables with severe data skew

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

When dividing the scanner, there is no longer a scanner for each
tablet, but according to the file size of the tablet, the tablet is
divided into multiple scanners for reading. By default, each scanner
reads 1GB. If the data is stored on a SATA disk, this value can be
increased, and if it is an SSD disk, the value can be decreased.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #skew) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -675,6 +676,62 @@ OLAPStatus Tablet::capture_rs_readers(const std::vector<Version>& version_path,
...
+    bool has_segment_range = false;
+    for (auto key_range : key_ranges) {
+        if (key_range->range_type == ScanRangeType::SEGMENT) {
+            for (const auto& s : key_range->segments) {
+                rowsets.emplace(s.first);
+            }
+            has_segment_range = true;
+        }
+    }
+    if (!has_segment_range) {
```
Comment (morningman) [inline be/src/olap/tablet.cpp:693]: Is it possible to have both KEY and SEGMENT key range in `key_ranges`?
If yes, we should clear the `rs_readers` before calling `capture_rs_readers()`

File: be/src/olap/tablet_meta.h [hunk 2]
```diff
@@ -285,6 +289,38 @@ inline void TabletMeta::set_cumulative_layer_point(int64_t new_point) {
     _cumulative_layer_point = new_point;
 }
 
+inline bool TabletMeta::contains_delete(const Version& version) const {
+    for (auto& rs : _rs_metas) {
```
Comment (morningman) [inline be/src/olap/tablet_meta.h:293]: I am not sure if rs in _rs_metas are ordered by version?

Comment (yangzhg) [reply be/src/olap/tablet_meta.h:293]: no need to order by version, if contains delete version this tablet will not split by segmnent

File: be/src/olap/tablet.cpp (hunk 1 above)
Comment (yangzhg) [reply be/src/olap/tablet.cpp:693]: if key_ranges  have SEGMENT key range, it will have KEY key range

File: be/src/olap/tablet.cpp [hunk 3]
```diff
@@ -675,6 +676,62 @@ OLAPStatus Tablet::capture_rs_readers(const std::vector<Version>& version_path,
...
+        return capture_rs_readers(spec_version, rs_readers, parent_tracker);
+    }
+    std::vector<Version> version_path;
+    RETURN_NOT_OK(capture_consistent_versions(spec_version, &version_path));
+    return capture_rs_readers_by_rowsets(version_path, rs_readers, rowsets, parent_tracker);
+}
+
+OLAPStatus Tablet::capture_rs_readers_by_rowsets(const std::vector<Version>& version_path,
+                                                 std::vector<RowsetReaderSharedPtr>* rs_readers,
+                                                 const std::set<RowsetId>& segments,
```
Comment (morningman) [inline be/src/olap/tablet.cpp:703]:
```suggestion
                                                 const std::set<RowsetId>& rowset_ids,
```

**Pull request closed without merging**